gradle/libs.versions.toml

```diff
@@ -1,7 +1,9 @@
 [versions]
 micronaut-docs = '2.0.0'
-micronaut = '4.0.0-M1'
+micronaut = '4.0.0-M2'
+micronaut-platform = '4.0.0-M1'
```

Comment:
> this was on the recommendation of @melix
> 
> basically I don't see a way to use the Micronaut Gradle plugin without specifying the platform

@melix is not enough to specify `micronaut-gradle-plugin = "4.0.0-M1"` why do we need platform version?

Reply:
Short answer: because we're hacking to use user build plugins in our module builds.

Long answer

Our modules use our internal build plugins. Those plugins **must not** depend on the platform, and automatically add a catalog named `mn` which corresponds to the **core** version of Micronaut. This plugin expects the `micronaut` property to refer to the **core** version.

 On the other hand, a user applying the Micronaut application plugin defines `micronautVersion` (`micronaut`, or `micronaut-platform` in a catalog), but there it corresponds to the **platform** version. Latest release also supports adding automatically a catalog named `mn`, but then it's the _platform_ catalog, unlike our internal build plugins.

Because in our tests we use the Gradle plugin, we are in a conflict situation where our internal plugins expect `micronaut` to be the version of **core**, while the user Gradle plugin expects it to be the **platform** version.

Therefore, we have to explain, in our test suites, that the `micronaut` version to be used, the platform one, is different. That we can do by explicitly having a variable which stores the platform version and uses it in the tests.